### DISCUSSION TITLE
Add color palette swapper

### DISCUSSION THREAD

**Discussion by TakashiOreki** [DesModder/DesModder#850] [Ideas]
Here's the console command if that makes it any easier:

`Calc.updateSettings({
    colors: {
        white : '#FFFFFF',gray_a : '#DDDDDD',gray_b : '#BBBBBB',gray_c : '#888888',gray_d : '#444444',gray_e : '#222222',black : '#000000',
        blue_a : '#C7E9F1',blue_b : '#9CDCEB',blue_c : '#58C4DD',blue_d : '#29ABCA',blue_e : '#236B8E',pure_blue : '#0000FF',
        teal_a : '#ACEAD7',teal_b : '#76DDC0',teal_c : '#5CD0B3',teal_d : '#55C1A7',teal_e : '#49A88F',
        green_a : '#C9E2AE',green_b : '#A6CF8C',green_c : '#83C167',green_d : '#77B05D',green_e : '#699C52',pure_green : '#00FF00',
        yellow_a : '#FFF1B6',yellow_b : '#FFEA94',yellow_c : '#FFFF00',yellow_d : '#F4D345',yellow_e : '#E8C11C',
        gold_a : '#F7C797',gold_b : '#F9B775',gold_c : '#F0AC5F',gold_d : '#E1A158',gold_e : '#C78D46',
        red_a : '#F7A1A3',red_b : '#FF8080',red_c : '#FC6255',red_d : '#E65A4C',red_e : '#CF5044',pure_red : '#FF0000',
        maroon_a : '#ECABC1',maroon_b : '#EC92AB',maroon_c : '#C55F73',maroon_d : '#A24D61',maroon_e : '#94424F',
        purple_a : '#CAA3E8',purple_b : '#B189C6',purple_c : '#9A72AC',purple_d : '#715582',purple_e : '#644172',
        pink : '#D147BD',light_pink : '#DC75CD',orange : '#FF862F',cyan:'#00FFFF',magenta:'#FF00FF',
        light_brown : '#CD853F',dark_brown : '#8B4513',gray_brown : '#736357',
    }
})`

**Comment by jared-hughes** [comment]
It would be nice to also allow a few other color palettes for a list of around 5 or so:
- default Desmos
- Manim
- Kelly (code at the bottom of this message)
- any other ideas?

For reference, the color order for Manim: 
![image](https://user-images.githubusercontent.com/20214911/139489835-90c54a8a-dee4-4dcd-95e6-8a34d77e148f.png)

Should reorder the colors first to avoid series breaking apart (such as blue on row 2), and maybe add two colors to bring it to a multiple of 6

And for Kelly:
![image](https://user-images.githubusercontent.com/20214911/139491476-90caa8f9-cbb0-4a0b-bfa7-b82f69784f24.png)

Should sort by hue, and add two colors (grays?) to bring it to a multiple of 6

Related: https://github.com/DesModder/DesModder/issues/76


Kelly's pallette (from https://stackoverflow.com/a/4382138/7481517)
```
Calc.updateSettings({colors: {
    black: "#000",
    white: "#FFF",
    vivid_yellow: "#FFB300",
    strong_purple: "#803E75",
    vivid_orange: "#FF6800",
    very_light_blue: "#A6BDD7",
    vivid_red: "#C10020",
    grayish_yellow: "#CEA262",
    medium_gray: "#817066",
    vivid_green: "#007D34",
    strong_purplish_pink: "#F6768E",
    strong_blue: "#00538A",
    strong_yellowish_pink: "#FF7A5C",
    strong_violet: "#53377A",
    vivid_orange_yellow: "#FF8E00",
    strong_purplish_red: "#B32851",
    vivid_greenish_yellow: "#F4C800",
    strong_reddish_brown: "#7F180D",
    vivid_yellowish_green: "#93AA00",
    deep_yellowish_brown: "#593315",
    vivid_reddish_orange: "#F13A13",
    dark_olive_green: "#232C16",
}})
```

**Comment by TakashiOreki** [comment]
So i changed it up so now its a perfect 12x6 on classic desmos and is also re-organized (had to add some more colors to complete it but here it is):
`
Calc.updateSettings({ colors: { 
pure_red : '#FF0000',red_e : '#CF5044',red_d : '#E65A4C',red_c : '#FC6255',red_b : '#FF8080',red_a : '#F7A1A3',
orange : '#FF862F', orange_a: '#FF8533', orange_b: '#FF954D', orange_c:'#FFA466', orange_d: '#FFB380', orange_e:'#FFC299',
gold_e : '#C78D46', gold_d : '#E1A158', gold_c : '#F0AC5F', gold_b : '#F9B775', gold_a : '#F7C797', gold_f:'#FBE3CB',
pure_yellow : '#FFFF00', yellow_f:'#F4E00E', yellow_e : '#E8C11C', yellow_d : '#F4D345',yellow_b : '#FFEA94',yellow_a : '#FFF1B6',
pure_green : '#00FF00', green_e : '#699C52', green_d : '#77B05D', green_c : '#83C167', green_b : '#A6CF8C', green_a : '#C9E2AE',
cyan:'#00FFFF', teal_e : '#49A88F', teal_d : '#55C1A7', teal_c : '#5CD0B3', teal_b : '#76DDC0', teal_a : '#ACEAD7',
pure_blue : '#0000FF', blue_e : '#236B8E', blue_d : '#29ABCA', blue_c : '#58C4DD', blue_b : '#9CDCEB',blue_a : '#C7E9F1',
pure_maroon:'#800000', maroon_e : '#94424F', maroon_d : '#A24D61', maroon_c : '#C55F73', maroon_b : '#EC92AB', maroon_a : '#ECABC1',
pure_purple:'#800080', purple_e : '#644172', purple_d : '#715582', purple_c : '#9A72AC', purple_b : '#B189C6', purple_a : '#CAA3E8',
magenta:'#FF00FF', pink_a:'#F018E9', pink_b:'#E02FD3', pink : '#D147BD',pink_c:'#D75EC5', light_pink : '#DC75CD',
dark_brown : '#8B4513', brown_a: '#7F5435', gray_brown : '#736357', brown_b: '#916E4F', brown_c: '#AF7A47', light_brown : '#CD853F',
black : '#000000', gray_d : '#444444', gray_c : '#888888', gray_b : '#BBBBBB', gray_a : '#DDDDDD', white : '#FFFFFF'
} })
`
![image](https://user-images.githubusercontent.com/31328512/140014995-4abe03f0-ee99-4e08-b32e-641b4a513042.png)

**Comment by jared-hughes** [comment]
Ooh, that is smooth. Is it a strict superset of the manim palette?

**Comment by jared-hughes** [comment]
Pixelpusher also suggests the colors from https://monokai.pro/.
